src/sentry/migrations/0372_backfill_perf_issue_events_issue_platform.py

```diff
@@ -0,0 +1,216 @@
+# Generated by Django 2.2.28 on 2023-02-14 19:42
+import uuid
+from datetime import datetime
+from typing import Any, Mapping, Sequence, Tuple
+
+from django.db import migrations
+from django.db.models import Count
+
+from sentry.event_manager import GroupInfo
+from sentry.eventstore.models import Event
+from sentry.issues.grouptype import (
+    PerformanceConsecutiveDBQueriesGroupType,
+    PerformanceFileIOMainThreadGroupType,
+    PerformanceMNPlusOneDBQueriesGroupType,
+    PerformanceNPlusOneAPICallsGroupType,
+    PerformanceNPlusOneGroupType,
+    PerformanceRenderBlockingAssetSpanGroupType,
+    PerformanceSlowDBQueryGroupType,
+    PerformanceUncompressedAssetsGroupType,
+)
+from sentry.issues.ingest import process_occurrence_data, send_issue_occurrence_to_eventstream
+from sentry.issues.issue_occurrence import IssueOccurrence, IssueOccurrenceData
+from sentry.issues.occurrence_consumer import lookup_event
+from sentry.new_migrations.migrations import CheckedMigration
+from sentry.snuba.dataset import Dataset, EntityKey
+from sentry.utils.performance_issues.performance_detection import EventPerformanceProblem
+
+
+def backfill_eventstream(apps, schema_editor):
+    """
+    Backfills Performance-issue events from the transaction table to the IssuePlatform dataset(search_issues).
+    1. Read all transactions that back a performance-issue(s) with the following criteria:
+        a. non-null group_ids column
+        b.
+    2. Prepare and format each transaction to be submitted through the eventstream.
+    3. Send it through the eventstream.
+    """
+
+    # TODO: need to make sure event_type is correct in eventstream.send()
+
+    Group = apps.get_model("sentry", "Group")
+    GroupHash = apps.get_model("sentry", "GroupHash")
+
+    project_perf_issues = (
+        Group.objects.all()
+        .filter(
+            type__in=(
+                PerformanceSlowDBQueryGroupType.type_id,
+                PerformanceRenderBlockingAssetSpanGroupType.type_id,
+                PerformanceNPlusOneGroupType.type_id,
+                PerformanceConsecutiveDBQueriesGroupType.type_id,
+                PerformanceFileIOMainThreadGroupType.type_id,
+                PerformanceNPlusOneAPICallsGroupType.type_id,
+                PerformanceMNPlusOneDBQueriesGroupType.type_id,
+                PerformanceUncompressedAssetsGroupType.type_id,
+            )
+        )
```

Comment:
We might need to temporarily (or permanently) add an index on `type` if we want this query to run fast. I feel like we've wanted it in the past, so maybe it makes sense to just add it.